### PR TITLE
Better errors for read

### LINKED ISSUE #3089: Reading a table as a matrixtable produces a bad error:
```hail.utils.java.FatalError: ClassCastException: is.hail.table.TableSpec cannot be cast to is.hail.variant.MatrixTableSpec```

### PR DESCRIPTION
**Pull request by tpoterba**
fixes #3089

### CODE REVIEW

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 1]
```diff
@@ -120,7 +122,10 @@ object FileFormat {
 object MatrixTable {
   def read(hc: HailContext, path: String,
     dropCols: Boolean = false, dropRows: Boolean = false): MatrixTable = {
-    val spec = RelationalSpec.read(hc, path).asInstanceOf[MatrixTableSpec]
+    val spec = (RelationalSpec.read(hc, path): @unchecked) match {
+      case mts: MatrixTableSpec => mts
+      case _: TableSpec => fatal(s"file is a Table, not a MatrixTable: '$path'")
```
Comment (jbloom22) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: ``path is to a...``

File: src/main/scala/is/hail/table/Table.scala [hunk 2]
```diff
@@ -69,11 +69,15 @@ object Table {
   }
 
   def read(hc: HailContext, path: String): Table = {
+
+    val spec = (RelationalSpec.read(hc, path): @unchecked) match {
+      case ts: TableSpec => ts
+      case _: MatrixTableSpec => fatal(s"file is a MatrixTable, not a Table: '$path'")
```
Comment (jbloom22) [inline src/main/scala/is/hail/table/Table.scala]: ``path is to a``

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 3]
```diff
@@ -41,6 +41,8 @@ object RelationalSpec {
     new MatrixTypeSerializer
 
   def read(hc: HailContext, path: String): RelationalSpec = {
+    if (!hc.hadoopConf.isDir(path))
+      fatal(s"a MatrixTable or Table file is a directory; path '$path' is not")
```
Comment (jbloom22) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: Tables and MatrixTables files are directories; path '$path' is not a directory.

File: src/main/scala/is/hail/variant/MatrixTable.scala (hunk 1 above)
Comment (tpoterba) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: I'm not sure this is clearer...

File: src/main/scala/is/hail/variant/MatrixTable.scala (hunk 3 above)
Comment (tpoterba) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: yes, this is clearer

**Pull request merged**